CP1404_practicals/prac_08/taxi_simulator.py

```diff
@@ -25,16 +25,34 @@ def main():
             print("Taxis available: ")
             taxis_available(taxi_types)
             taxi_choice = int(input("choose taxi: "))
-            taxi_chosen = taxi_types[taxi_choice]
-            taxis_available(taxi_types)
+            try:
+                current_taxi = taxi_types[taxi_choice]
+            except IndexError:
+                print("Invalid choice")
         elif menu_choice == "d":
-            print("lets drive taxi")
+            if current_taxi:
+                current_taxi.start_fare()
+                driving_distance = float(input("Drive how far? "))
+                current_taxi.drive(driving_distance)
+                cost = current_taxi.get_fare()
+                print("Your {} trip cost you ${:.2f}".format(current_taxi.name, cost))
+                total_cost += cost
+            else:
+                print("you need to choose a taxi before you can drive")
+        else:
+            print("Invalid option")
+        print("Bill to date: ${:.2f}".format(total_cost))
+        print(MENU)
+        menu_choice = input(">>> ").lower()
+    print("Bill to date: ${:.2f}".format(total_cost))
+    print("Taxis are now:")
+    taxis_available(taxi_types)
```

Comment:
you can condense these using string formatting eg. print("Taxis are now:\n{}.format(taxis_available(taxi_types))